reascripts/ReaSpeech/source/ReaSpeechWorker.lua

```diff
@@ -145,39 +145,51 @@ function ReaSpeechWorker:handle_request(request)
   self:log('Processing speech...')
   self.job_count = #request.jobs
 
-  local data = {
-    task = request.translate and 'translate' or 'transcribe',
-    output = 'json',
-    use_async = 'true',
-    vad_filter = request.vad_filter and 'true' or 'false',
-    word_timestamps = 'true',
-    model_name = request.model_name,
-  }
-
-  if request.language and request.language ~= '' then
-    data.language = request.language
-  end
-
-  if request.hotwords and request.hotwords ~= '' then
-    data.hotwords = request.hotwords
-  end
-
-  if request.initial_prompt and request.initial_prompt ~= '' then
-    data.initial_prompt = request.initial_prompt
+  for _, job in ipairs(self:expand_jobs_from_request(request)) do
+    table.insert(self.pending_jobs, job)
   end
+end
 
+function ReaSpeechWorker:expand_jobs_from_request(request)
+  local jobs = {}
   local seen_path = {}
   for _, job in pairs(request.jobs) do
     if not seen_path[job.path] then
       seen_path[job.path] = true
-      table.insert(self.pending_jobs, {
+      local uploads = {}
+      if request.file_uploads then
+        for key, path_or_path_function in pairs(request.file_uploads) do
+          if type(path_or_path_function) == 'function' then
+            uploads[key] = path_or_path_function(job)
+          else
+            uploads[key] = path_or_path_function
+          end
+        end
+      end
+      table.insert(jobs, {
         job = job,
         endpoint = request.endpoint,
-        data = data,
+        data = request.data,
+        file_uploads = uploads,
         callback = request.callback
       })
     end
   end
+
+  if #jobs > 0 then
+    return jobs
+  end
+
+  for _, _ in pairs(request.file_uploads) do
```

Comment:
if we got this far, we're not dealing with the item/take/path "jobs" and assume that if there's an entry in `file_uploads` that this is sufficient.